htlcswitch/hop/payload.go

```diff
@@ -157,14 +164,22 @@ func NewPayloadFromReader(r io.Reader) (*Payload, error) {
 		mpp = nil
 	}
 
+	for t, parseResult := range parsedTypes {
```

Comment:
Feels like there's another way to do this with less copying.

Reply:
In line with comment above, I don't think it matters much. If you want the tlv stream to return a map that we can use directly, we need to add more controls to the stream to specify what exactly we expect.

Reply:
suggestion from @cfromknecht: delete items from the map

Reply:
It seems it isn't possible to cast `map[Type][]byte` to `map[uint64][]byte` though

Reply:
Talked about this offline with @roasbeef. We do want to stop `tlv.Type` from being used anywhere. Copying the map (but not the values) looks to be ok.